### PR TITLE
PLAT-1040 | Add laravel schedule monitoring for LDEK

### PR DESCRIPTION
**Pull request by kvas-damian**
Don't merge without: https://github.com/bethinkpl/ansible/pull/52

### CODE REVIEW

File: app/Console/Commands/CommandWithMonitoring.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+			if ($metricName) {
+				(new PrometheusPushgateway())->notify(env('MONITORING_METRIC_NAME'));
+			} else {
+				\Log::debug('Monitoring notification silenced on non-production env');
+			}
+		}
+		return $result;
+	}
+
+	abstract public function handleBody();
```
Comment (bognix) [inline app/Console/Commands/CommandWithMonitoring.php]: handleBody sounds odd a little bit. How about execute?

File: app/Console/Kernel.php [hunk 2]
```diff
@@ -33,7 +33,7 @@ protected function schedule(Schedule $schedule)
 			->hourly()
 			->withoutOverlapping()
 			->after(function () {
-				$this->notifyPrometheusPushgateway('orders_stats_export');
+				(new PrometheusPushgateway())->notify('orders_stats_export');
```
Comment (bognix) [inline app/Console/Kernel.php]: Does it mean that if I run one of those commands using `php artisan...` they will report to Prometheus? Maybe we can add some flag to disable reporting in such case? I think the reporting shouldn't be polluted with "manual" executions if not stated explicitly.

File: app/Console/PrometheusPushgateway.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+namespace App\Console;
+
+
+use GuzzleHttp\Exception\GuzzleException;
+
+class PrometheusPushgateway
+{
+	public function notify($metricName)
+	{
+		$client = new \GuzzleHttp\Client();
```
Comment (bognix) [inline app/Console/PrometheusPushgateway.php]: could you keep the `use` consistent? Either declare `use ...` at the top of the file or use the absolute names.

File: config/app.php [hunk 4]
```diff
@@ -54,6 +54,16 @@
 
 	'url' => env('APP_URL', 'http://localhost'),
 
+	/*
+	|--------------------------------------------------------------------------
+	| Application Instance Name
+	|--------------------------------------------------------------------------
+	|
+	| Identify different proction environments like lek, ldek, etc...
```
Comment (rogatty) [inline config/app.php]: `proction`?

File: app/Console/Commands/StoreProgress.php [hunk 5]
```diff
@@ -6,11 +6,10 @@
 use App\Models\UserCourseProgress;
 use Closure;
 use Exception;
-use Illuminate\Console\Command;
 use Illuminate\Support\Facades\DB;
 use Illuminate\Support\Facades\Redis;
 
-class StoreProgress extends Command
+class StoreProgress extends CommandWithMonitoring
```
Comment (rogatty) [inline app/Console/Commands/StoreProgress.php]: Remember to update the rest of the commands too.

File: app/Console/Kernel.php (hunk 2 above)
Comment (rogatty) [reply app/Console/Kernel.php]: If you run a command using artisan, you don't go through this flow. You go through `CommandWithMonitoring::handle()` which has a check for `MONITORING_METRIC_NAME` which is set only in k8s cron jobs.

This here is a temporary solution until we migrate LEK production to k8s. Then, we can remove all these Laravel schedule jobs.

File: app/Console/Commands/StoreProgress.php (hunk 5 above)
Comment (kvas-damian) [reply app/Console/Commands/StoreProgress.php]: Skleroza...

File: app/Console/Commands/CommandWithMonitoring.php (hunk 1 above)
Comment (kvas-damian) [reply app/Console/Commands/CommandWithMonitoring.php]: Execute is already reserved: https://github.com/laravel/framework/blob/5.8/src/Illuminate/Console/Command.php#L181
Do you have other names in your mind?

Comment (rogatty) [reply app/Console/Commands/CommandWithMonitoring.php]: `run`

Comment (kvas-damian) [reply app/Console/Commands/CommandWithMonitoring.php]: https://github.com/laravel/framework/blob/5.8/src/Illuminate/Console/Command.php#L163

**Comment by kvas-damian** [comment]
LGTM

**Pull request merged**